src/components/Tags/index.tsx

```diff
@@ -0,0 +1,282 @@
+  color: ${({ theme: { colors } }) => colors.gray700};
+  border: none;
+  outline: none;
+  background-color: ${({ theme: { colors } }) => colors.white};
+  &::placeholder {
+    color: ${({ theme: { colors } }) => colors.gray550};
+  }
+`
+
+const convertTagArrayToTagStateArray = (tags: TagsProp) =>
+  tags.map((tag, index) =>
+    typeof tag === 'object'
+      ? { ...tag, index: getUUID(`tag-${index}`) }
+      : { index: getUUID(`tag-${index}`), label: tag },
+  )
+
+type TagsProp = (string | { label?: string | null } | undefined | null)[]
+interface TagsProps {
+  areTagsObject?: boolean
+  disabled?: boolean
+  id?: string
+  manualInput?: boolean
+  name?: string
+  onChange?: ChangeEventHandler<HTMLInputElement>
+  onChangeError?: ChangeEventHandler<HTMLInputElement>
+  placeholder?: string
+  tags?: TagsProp
+  variant?: Variant
+}
+
+const Tags: VoidFunctionComponent<TagsProps> = ({
+  areTagsObject = false,
+  disabled = false,
+  id,
+  manualInput = true,
+  name,
+  onChange,
+  onChangeError,
+  placeholder,
+  tags = [],
+  variant = 'base',
+  ...props
+}) => {
+  const [tagsState, setTags] = useState(convertTagArrayToTagStateArray(tags))
+  const [input, setInput] = useState<string>('')
+  const [status, setStatus] = useState<{ [key: string]: StatusValue }>({})
+
+  useEffect(() => {
+    setTags(convertTagArrayToTagStateArray(tags))
+  }, [tags])
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const dispatchOnChange = async (newState: TagsProp) =>
+    onChange?.(
+      areTagsObject
+        ? newState.map(({ index, ...tag }) => tag)
+        : newState.map(({ label }) => label),
+    )
+
+  const handleContainerClick = () => {
+    if (inputRef.current) {
+      inputRef?.current?.focus()
+    }
+  }
+
+  const onInputChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setInput(e.target.value)
+
+  const addTag = async (e?: FocusEvent<HTMLDivElement>) => {
+    const newTags = input
+      ? [...tagsState, { index: getUUID('tag'), label: input }]
+      : tagsState
+    setInput('')
+    setTags(newTags)
+    if (newTags.length !== tagsState.length) {
+      setStatus({ [newTags[newTags.length - 1].index]: STATUS.LOADING })
+    }
+    try {
+      await dispatchOnChange(newTags)
+      setStatus({ [newTags[newTags.length - 1].index]: STATUS.IDLE })
+    } catch (e) {
+      onChangeError?.(e)
+      setTags(tagsState)
+    }
+  }
+
+  const deleteTag = async (tagIndex: string) => {
+    setStatus({ [tagIndex]: STATUS.LOADING })
+    const index = tagsState.findIndex(({ index }) => index === tagIndex)
+    const newTags = [...tagsState]
+    newTags.splice(index, 1)
+    try {
+      await dispatchOnChange(newTags)
+      setTags(newTags)
+      setStatus({ [tagIndex]: STATUS.IDLE })
+    } catch (e) {
+      onChangeError?.(e)
+      setTags(tagsState)
+    }
+  }
+
+  const handleInputKeydown = e => {
+    // 32 = Space | 13 = Enter | 8 = backspace
+    if (e.keyCode === 32 || e.keyCode === 13) {
```

Comment:
keyCode is deprecated, is it possible to change it in this PR ?

Reply:
i made something for this component, but i believe we should probably use https://github.com/timoxley/keycode